integrations/erc4337/userop.go

```diff
@@ -0,0 +1,91 @@
+package erc4337
+
+import (
+	"context"
+
+	"github.com/indexsupply/x/contrib/erc4337"
+	"github.com/indexsupply/x/e2pg"
+
+	"github.com/jackc/pgx/v5"
+)
+
+type integration struct {
+	name string
+}
+
+var Integration = integration{
+	name: "ERC4337 UserOperationEvent",
+}
+
+func (i integration) Events(ctx context.Context) [][]byte {
+	return [][]byte{erc4337.UserOperationEventSignatureHash}
+}
+
+func (i integration) Delete(ctx context.Context, pg e2pg.PG, h []byte) error {
+	const q = `
+		delete from erc4337_userops
+		where task_id = $1
+		and chain_id = $2
+		and block_hash = $3
+	`
+	_, err := pg.Exec(ctx, q, e2pg.TaskID(ctx), e2pg.ChainID(ctx), h)
+	return err
+}
+
+func (i integration) Insert(ctx context.Context, pg e2pg.PG, blocks []e2pg.Block) (int64, error) {
+	var rows = make([][]any, 0, 1<<12) // 2^12 max batch size?
```

Comment:
There isn't a max. I over allocated the slice to avoid allocations inside the loop. In the future I would like to coalesce this code so that integrations don't have to consider these kinds of things. But we don't need to take action in this pr.

Reply:
Gotcha, good to know! Deleted the comment so it doesn't create confusion for future readers.